src/wallet/coincontrol.cpp

```diff
@@ -88,6 +88,20 @@ std::optional<uint32_t> CCoinControl::GetSequence(const COutPoint& outpoint) con
     return it->second.GetSequence();
 }
 
+bool CCoinControl::HasScripts(const COutPoint& outpoint) const
+{
+    const auto it = m_selected.find(outpoint);
+    if (it == m_selected.end()) {
+        return false;
+    }
+    return it->second.HasScripts();
```

Comment:
(in commit 6fcbf9f44358f85aab4c0b6617c113b4c37da5a8)
nit: could be written as a one-liner
```suggestion
    return (it != m_selected.end() && it->second.HasScripts());
```

Reply:
Done